test/Microsoft.AspNetCore.Authentication.Test/VirtualHandlerTests.cs

```diff
@@ -277,21 +280,65 @@ public class VirtualHandlerTests
             Assert.Equal(0, handler2.SignInCount);
         }
 
+        [Fact]
+        public async Task TargetsSelfResultsInFallbackToDefault()
+        {
+            var services = new ServiceCollection().AddOptions().AddLogging();
+
+            services.AddAuthentication(o =>
+            {
+                o.AddScheme<TestHandler>("auth1", "auth1");
+                o.DefaultScheme = "auth1";
+            })
+            .AddScheme("forward", "forward", p =>
+            {
+                p.ForwardDefault = "forward";
```

Comment:
This looks invalid and should throw as discussed in the virtual handler.

Reply:
Yeah this might not be needed anymore now that the virtual logic has been merged into all handlers

Reply:
But if we go that route, we probably need to always require a ForwardDefault being set for a pure virtual handler scheme

Reply:
Or a selector.

Reply:
Nah the selector doesn't guarantee a fallback, since the selector can return null, I think we need to force a default target which we can do via the AddScheme overload, since this type exists and has different behavior, we should probably just go back to calling it AddVirtualScheme("virtualScheme", "displayName", "defaultTarget")

Reply:
Or throw if anything ever actually reaches the virtual handler's HandleX methods.

Reply:
We can discuss this more in the design review, but as soon as we force a default forward for any virtual scheme, there actually is no reason to even support pure virtual schemes.  They can just use whatever the default forward and configure that as desired, the only slight difference would be if they wanted to use a non cookie default and attempt to forward something like sign-in which isn't always supported.

Reply:
AuthSamples.PathBasedSelection would simply become:

```
            services.AddAuthentication("DynamicCookie")
                .AddScheme<AuthenticationSchemeOptions, ApiAuthHandler>("Api", o => { })
                .AddCookie("DynamicCookie", options =>
                {
                    o.DefaultSelector = ctx => ctx.Request.Path.StartsWithSegments("/api") ? "Api" : "DynamicCookie";
                    options.AccessDeniedPath = "/account/denied";
                    options.LoginPath = "/account/login";
                })
```